jwst/barshadow/bar_shadow.py

```diff
@@ -11,94 +11,109 @@
             shutter_status = slitlet.shutter_state
             if len(shutter_status) > 0:
                 shadow = create_shadow(shutter_elements, shutter_status)
-                #
-                # For each pixel in the slit subarray
-                #   Make a grid of indices for pixels in the subarray
-                x, y = wcstools.grid_from_bounding_box(slitlet.meta.wcs.bounding_box, step=(1,1))
-                #   Create the transformation from slit_frame to detector
+
+                # For each pixel in the slit subarray,
+                # make a grid of indices for pixels in the subarray
+                x, y = wcstools.grid_from_bounding_box(slitlet.meta.wcs.bounding_box, step=(1, 1))
+
+                # Create the transformation from slit_frame to detector
                 det2slit = slitlet.meta.wcs.get_transform('detector', 'slit_frame')
-                #   Use this transformation to calculate x, y and wavelength
+
+                # Use this transformation to calculate x, y, and wavelength
                 xslit, yslit, wavelength = det2slit(x, y)
-                #   The returned y values are scaled to where the slit height is 1
+
+                # The returned y values are scaled to where the slit height is 1
                 # (i.e. a slit goes from -0.5 to 0.5).  The barshadow array is scaled
-                # so that the separation between the slit centers is 1, i.e. slit height
-                # + interslit bar
+                # so that the separation between the slit centers is 1,
+                # i.e. slit height + interslit bar
                 yslit = yslit / SLITRATIO
-                #   Convert the Y and wavelength to a pixel location
-                #   in the  bar shadow array
+
+                # Convert the Y and wavelength to a pixel location in the  bar shadow array
                 index_of_fiducial = shutter_status.find('x')
-                #
+
                 # The shutters go downwards, i.e. the first shutter in shutter_status corresponds to
                 # the last in the shadow array.  So the center of the first shutter referred to in
                 # shutter_status has an index of shadow.shape[0] - shutter_height.  Each subsequent
                 # shutter center has an index shutter_height greater.
                 index_of_fiducial_in_array = shadow.shape[0] - shutter_height * (1 + index_of_fiducial)
                 yrow = index_of_fiducial_in_array - yslit * shutter_height
                 wcol = (wavelength - w0)/wave_increment
+
                 # Interpolate the bar shadow correction for non-Nan pixels
                 correction = interpolate(yrow, wcol, shadow)
-                # Add the correction array and variance to the datamodel
+
+                # Add the correction array to the datamodel
                 slitlet.barshadow = correction
+
+                # Apply the correction to the science and uncertainty arrays
+                slitlet.data /= correction
```

Comment:
What is it that sets the shape of `correction` to be the same as `slitlet.data`?  Is it the `bounding_box`?  If so, is it a reasonable assumption that they are the same size in each dimension?

Reply:
The size of the correction array is set in the ``interpolate`` function, based on the size of the input array.